test/e2e/storage/external/external.go

```diff
@@ -0,0 +1,217 @@
+}
+
+func (t testDriverParameter) Set(filename string) error {
+	driver, err := t.loadDriverDefinition(filename)
+	if err != nil {
+		return err
+	}
+
+	Describe("External Storage "+testsuites.GetDriverNameWithFeatureTags(driver), func() {
+		f := framework.NewDefaultFramework(driver.ShortName)
+
+		BeforeEach(func() {
+			// Reset config. The driver might have modified its copy
+			// in a previous test.
+			driver.GetDriverInfo().Config = testsuites.TestConfig{
+				Prefix:    driver.ShortName,
+				Framework: f,
+			}
+
+			// setupDriver
+			driver.CreateDriver()
+		})
+
+		AfterEach(func() {
+			// Cleanup driver
+			driver.CleanupDriver()
+		})
+
+		testsuites.RunTestSuite(f, driver, csiTestSuites,
+			func(patterns []testpatterns.TestPattern) []testpatterns.TestPattern {
+				return patterns
+			})
+	})
+
+	return nil
+}
+
+func (t testDriverParameter) loadDriverDefinition(filename string) (*driverDefinition, error) {
+	if filename == "" {
+		return nil, errors.New("missing file name")
+	}
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	// Some reasonable defaults follow.
+	driver := &driverDefinition{
+		DriverInfo: testsuites.DriverInfo{
+			SupportedFsType: sets.NewString(
+				"", // Default fsType
+			),
+		},
+		ClaimSize: "5Gi",
+	}
+	// TODO: strict checking of the file content once https://github.com/kubernetes/kubernetes/pull/71589
+	// or something similar is merged.
+	if err := runtime.DecodeInto(legacyscheme.Codecs.UniversalDecoder(), data, driver); err != nil {
+		return nil, errors.Wrap(err, filename)
+	}
+	return driver, nil
+}
+
+var _ testsuites.TestDriver = &driverDefinition{}
+
+// We have to implement the interface because dynamic PV may or may
+// not be supported. driverDefinition.SkipUnsupportedTest checks that
+// based on the actual driver definition.
+var _ testsuites.DynamicPVTestDriver = &driverDefinition{}
+
+// Needed only for deserialization which isn't done, therefore the
+// functions below can be empty stubs.
+var _ runtime.Object = &driverDefinition{}
+
+// DriverDefinition needs to be filled in via a .yaml or .json
+// file. It's methods then implement the TestDriver interface, using
+// nothing but the information in this struct.
+type DriverDefinition struct {
+	// DriverInfo is the static information that the storage testsuite
+	// expects from a test driver. See test/e2e/storage/testsuites/testdriver.go
+	// for details. The only field with a non-zero default is the list of
+	// supported file systems (SupportedFsType): it is set so that tests using
+	// the default file system are enabled.
+	DriverInfo testsuites.DriverInfo
+
+	// ShortName is used to create unique names for test cases and test resources.
+	ShortName string
+
+	// StorageClass is required for enabling dynamic provisioning tests.
+	// When set to "default", a StorageClass with DriverInfo.Name as provisioner is used.
+	// Everything else is treated like the filename of a .yaml or .json file
```

Comment:
Strictly speaking, it depends on how the `e2e/framework/testfiles` package is set up in the E2E test suite that includes this package here. But you are right, I need to mention that here and explain the relationship with the `-repo-root` parameter.

A safer approach will be to use an absolute file. https://github.com/kubernetes/kubernetes/pull/71667 was merged, so that works.